sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowConverters.scala

```diff
@@ -128,6 +128,92 @@ private[sql] object ArrowConverters extends Logging {
     }
   }
 
+  private[sql] def toArrowBatchIterator(
```

Comment:
can we simply change `toBatchIterator` to return row count as well? The perf overhead is very small but the maintenance overhead is much larger if we have 2 similar methods.

Reply:
`toArrowBatchIterator` also write schema before each record batch, while `toBatchIterator ` just output record batch.

and I am going to update `toArrowBatchIterator` in a follow-up to control each batch size < 4MB as per the suggestions https://github.com/apache/spark/pull/38468#discussion_r1018951362   https://github.com/apache/spark/pull/38468#discussion_r1013186031

I think we can deduplicate the codes then

Reply:
I see, we probably need a better naming, as it's hard to tell the difference between `toBatchIterator` and `toArrowBatchIterator`.